libs/application/templates/passport/src/lib/PassportTemplate.ts

```diff
@@ -6,84 +6,113 @@ import {
+import { Features } from '@island.is/feature-flags'
+import { m } from '../lib/messages'
 
 const PassportTemplate: ApplicationTemplate<
   ApplicationContext,
   ApplicationStateSchema<Events>,
   Events
 > = {
   type: ApplicationTypes.PASSPORT,
-  name: 'Umsókn um vegabréf',
+  name: m.formName.defaultMessage,
+  featureFlag: Features.passportApplication,
   dataSchema,
   stateMachineConfig: {
-    initial: 'draft',
+    initial: States.DRAFT,
     states: {
       draft: {
         meta: {
-          name: 'Umsókn um vegabréf',
+          name: m.formName.defaultMessage,
           progress: 0.33,
-          lifecycle: DefaultStateLifeCycle,
+          lifecycle: {
+            shouldBeListed: false,
+            shouldBePruned: true,
+            whenToPrune: 24 * 3600 * 1000,
+          },
+          onExit: {
+            apiModuleAction: ApiActions.checkForDiscount,
+          },
           roles: [
             {
-              id: 'applicant',
+              id: Roles.APPLICANT,
               formLoader: () =>
                 import('../forms/Draft').then((val) =>
                   Promise.resolve(val.Draft),
                 ),
               actions: [
-                { event: 'SUBMIT', name: 'Staðfesta', type: 'primary' },
+                {
+                  event: DefaultEvents.SUBMIT,
+                  name: m.confirm.defaultMessage,
+                  type: 'primary',
+                },
+                {
+                  event: DefaultEvents.PAYMENT,
+                  name: m.confirm.defaultMessage,
+                  type: 'primary',
+                },
               ],
               write: 'all',
+              delete: true,
             },
           ],
         },
         on: {
-          SUBMIT: {
-            target: 'approved',
+          [DefaultEvents.SUBMIT]: { target: States.DRAFT },
+          [DefaultEvents.PAYMENT]: { target: States.PAYMENT },
+        },
+      },
+      [States.PAYMENT]: {
+        meta: {
+          name: 'Payment state',
+          actionCard: {
+            description: m.payment,
           },
+          progress: 0.9,
+          lifecycle: DefaultStateLifeCycle,
+          onEntry: {
+            apiModuleAction: ApiActions.createCharge,
+          },
+          roles: [
+            {
+              id: Roles.APPLICANT,
+              formLoader: () =>
+                import('../forms/Payment').then((val) =>
+                  Promise.resolve(val.payment),
+                ),
+              actions: [
+                { event: DefaultEvents.SUBMIT, name: '', type: 'primary' },
+              ],
+              write: 'all',
+            },
+          ],
+        },
+        on: {
+          [DefaultEvents.SUBMIT]: { target: States.DONE },
         },
       },
-      approved: {
+      [States.DONE]: {
         meta: {
-          name: 'Approved',
+          name: 'Done',
           progress: 1,
           lifecycle: DefaultStateLifeCycle,
```

Comment:
Should we not prune the applicatation at some point when we are done. Is there a reason for us to hold on to the datal.